### ISSUE TITLE
better error handling when hostname does not match

### ISSUE THREAD

**Issue by FrancYescO** [thekuwayama/nslookupot#12]
when the hostname does not match the certificate we get a full traceback... maybe some cleanup is needed

```
Traceback (most recent call last):
	6: from /usr/local/bin/nslookupot:23:in `<main>'
	5: from /usr/local/bin/nslookupot:23:in `load'
	4: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/exe/nslookupot:8:in `<top (required)>'
	3: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/cli.rb:99:in `run'
	2: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/resolver.rb:40:in `resolve_resources'
	1: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/resolver.rb:89:in `gen_sock'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/openssl/ssl.rb:395:in `post_connection_check': hostname "cloudflare-dns.com" does not match the server certificate (OpenSSL::SSL::SSLError)
```

i also got this error while debugging a dns server behind traefik:
```
Traceback (most recent call last):
	6: from /usr/local/bin/nslookupot:23:in `<main>'
	5: from /usr/local/bin/nslookupot:23:in `load'
	4: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/exe/nslookupot:8:in `<top (required)>'
	3: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/cli.rb:99:in `run'
	2: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/resolver.rb:40:in `resolve_resources'
	1: from /Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/resolver.rb:88:in `gen_sock'
/Library/Ruby/Gems/2.6.0/gems/nslookupot-0.0.9/lib/nslookupot/resolver.rb:88:in `connect': SSL_connect returned=1 errno=0 state=SSLv3 read server hello A: (null) (OpenSSL::SSL::SSLError)
```

**Comment by thekuwayama** [comment]
Thank you for your comments!
Solved by #13 . 👍 

```sh-session
$ bundle exec exe/nslookupot --server=8.8.8.8 crypto.cloudflare.com
Server:         8.8.8.8
Address:        8.8.8.8#853

** DoT Server Unavailable: hostname "cloudflare-dns.com" does not match the server certificate
```

**Comment by thekuwayama** [comment]
@FrancYescO 
[nslookupot-0.0.10](https://rubygems.org/gems/nslookupot/versions/0.0.10) is released. You can use it! :+1:
